### PR TITLE
Cleanup CMake files and build scripts for Android

### PR DESCRIPTION
**Pull request by Maratyszcza**
- Remove USE_ARM64 option because it doesn't do what is expected
- Disable ARM ComputeLibrary for non-ARM/ARM64 builds
- Remove analysis of CMake options from scripts/build_android.sh
- Add user-specified CMake options at the end of command line to allow overriding defaults
- Update README for ARM ComputeLibrary integration and do not require to disable NNPACK for ARM64 build with ARM ComputeLibrary

I checked that Caffe2 with ARM ComputeLibrary builds for both ARMv7 and ARM64

### CODE REVIEW

**Comment by Yangqing** [comment]
It looks all good. @jerryzh168 do you mind taking a look?

Also, contbuild is having hiccups so we might want to wait and retest. Let me try. @caffe2bot  retest this please.

File: caffe2/CMakeLists.txt [hunk 1]
```diff
@@ -97,14 +97,16 @@ set(Caffe2_MAIN_LIBS)
 # Compile exposed libraries.
 add_library(caffe2 ${Caffe2_CPU_SRCS} $<TARGET_OBJECTS:Caffe_PROTO> $<TARGET_OBJECTS:Caffe2_PROTO>)
 if(USE_ACL)
-  if(NOT USE_ARM64)
+  # Check if 32-bit ARM: armv7, armv7-a, armv7l, etc
```
Comment (Yangqing) [inline caffe2/CMakeLists.txt:100]: Do you mind moving this part to cmake/Dependencies.cmake?

File: scripts/build_android.sh [hunk 2]
```diff
@@ -75,38 +75,16 @@ if [ "${VERBOSE:-}" == '1' ]; then
...
 CMAKE_ARGS+=("-DANDROID_NATIVE_API_LEVEL=21")
 CMAKE_ARGS+=("-DANDROID_CPP_FEATURES=rtti exceptions")
 # TODO: As the toolchain file doesn't support NEON-FP16 extension,
 # we disable USE_MOBILE_OPENGL for now, it will be re-enabled in the future.
 CMAKE_ARGS+=("-DUSE_MOBILE_OPENGL=OFF")
 
-# Compiler flags
-CMAKE_ARGS+=("-DCMAKE_C_FLAGS=")
-if $USE_GCC;then
-  CMAKE_ARGS+=("-DCMAKE_CXX_FLAGS=-s")
```
Comment (jerryzh168) [inline scripts/build_android.sh:108]: Are we agreed to remove this, if so, probably write down the reasons here as well for reference. Thanks!

**Review by jerryzh168 — APPROVED** [review]
Thanks for fixing the build scripts!

**Comment by Yangqing** [comment]
Android is passing so we should probably be good to go :)

**Review by Yangqing — APPROVED** [review]

**Pull request merged**